xformers/components/attention/csrc/cuda/attention.cu

```diff
@@ -669,21 +882,43 @@ __global__ void attention_backward_grad_v_kernel(
     gb[q_item_idx] =
         reinterpret_cast<vec_t*>(grad_out[batch_idx][index].data());
   }
-
   for (int64_t i = 0; i < TILE_SIZEQ; i++) {
     int64_t index = query_idx + i - kBlockSizeQ * threadIdx.x;
     if (check_bounds)
       index = min(index, M - 1);
+    qbb[i] = reinterpret_cast<vec_t*>(query[batch_idx][index].data());
     gbb[i] = reinterpret_cast<vec_t*>(grad_out[batch_idx][index].data());
   }
 
+  for (int64_t i = 0; i < TILE_SIZEK; i++) {
+    int64_t index = l + i - kBlockSizeK * threadIdx.y;
+    if (check_bounds)
+      index = min(index, N - 1);
+    kbb[i] = reinterpret_cast<vec_t*>(key[batch_idx][index].data());
+  }
+
   for (int i = 0; i < kBlockSizeQ; i++) {
     int64_t index = query_idx + i;
     if (check_bounds)
       index = min(index, M - 1);
     normalizer[i] = logsumexp_normalizer[batch_idx][index];
   }
 
+  for (int q_item_idx = 0; q_item_idx < kBlockSizeQ; q_item_idx++) {
+    int64_t index = query_idx + q_item_idx;
+    if (index >= M)
+      break;
+
+    auto out_i = reinterpret_cast<vec_t*>(output[batch_idx][index].data());
```

Comment:
nit: would 'const vec_t *' make sense here, read only ? may not be idiomatic cuda, in c++ it's quite typical in some codebases to be a little strict around that